python/cudf/cudf/core/column/categorical.py

```diff
@@ -54,21 +45,45 @@ def as_ordered(self, **kwargs):
         if data is not None:
             from cudf import Series
 
-            return Series(data=data)
+            parent = self._parent
+            dtype = CategoricalDtype(
+                categories=parent.dtype.categories, ordered=True,
+            )
+            return Series(
+                column.build_column(
+                    data=None,
+                    dtype=dtype,
+                    mask=parent.mask,
+                    name=parent.name,
+                    children=parent.children,
+                )
+            )
```

Comment:
Doesn't maintain name / index again. It didn't previously though either.

Reply:
This function feels super weird because it's at the column level but returns a Series...